cpp/src/tracing.cpp

```diff
@@ -0,0 +1,255 @@
+/*
```

Comment:
This file should be renemaed tracing_opentelemetry.cpp as it is the Opentelemetry implementation of tracing - in theory there could be others.